spec/models/manageiq/providers/amazon/cloud_manager/vcr_specs/targeted_refresh_spec.rb

```diff
@@ -440,6 +440,79 @@
           assert_counts(expected_counts)
         end
       end
+
+      it "will refresh a volume with volume_snapshot" do
+        base_volume = ManagerRefresh::Target.new(
+          :manager_id  => @ems.id,
+          :association => :cloud_volumes,
+          :manager_ref => {
+            :ems_ref => "vol-0e1613cacf4688009"
+          }
+        )
+
+        volume = ManagerRefresh::Target.new(
+          :manager_id  => @ems.id,
+          :association => :cloud_volumes,
+          :manager_ref => {
+            :ems_ref => "vol-0e4c86c12b28cead8"
+          }
+        )
+
+        snapshot = ManagerRefresh::Target.new(
+          :manager_id  => @ems.id,
+          :association => :cloud_volume_snapshots,
+          :manager_ref => {
+            :ems_ref => "snap-055095f47fab5e749"
+          }
+        )
+
+        2.times do # Run twice to verify that a second run with existing data does not change anything
+          @ems.reload
+
+          VCR.use_cassette(described_class.name.underscore + "_targeted/cloud_volume_with_snapshot") do
+            EmsRefresh.refresh([base_volume, volume, snapshot])
+          end
+          @ems.reload
+
+          assert_specific_cloud_volume_vm_on_cloud_network
+          assert_specific_cloud_volume_snapshot
+
+          expected_counts = {
+            :auth_private_key              => 0,
```

Comment:
I think another improvement would be, if missing keys would default to `0` this would shrink those execpected_counts